File: collatz/julia/countCollatz.jl
#!/usr/bin/julia

# Count sequence in Collatz Conjecture
function CountCollatz(n)
      counter = 1
      while n > 1
            println(Int64(n))
            if(n % 2 == 0)
                  n = n / 2
            else
                  n = n * 3 + 1
            end
            counter += 1
      end
      println(Int64(n))
      return counter
end;

# Main Funciton
print("Enter a number : ")
userInput = parse(Int64, readline())
println("Numbers in the Collatz Conjecture : ", string(CountCollatz(userInput)))
exit(0)

File: collatz/go/countCollatz.go

// Count the total number of numbers for a number in the collatz conjecture

package main

import("fmt")

func main() { 

      var userInput int
      fmt.Print("Enter a number : ")
      fmt.Scan(&userInput)
      fmt.Println("Numbers in the Collatz Sequence : ",  CountCollatz(userInput))
      
}

func CountCollatz(n int) int{ 
    
    var counter int = 1
    for n > 1 { 
          fmt.Println(n)
          if(n % 2 == 0) { 
                n = n / 2
          } else  { 
                n = n * 3 + 1
          }
          counter++
    }
    fmt.Println(n)
    return counter
    
}